src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -570,25 +570,26 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             BoundLambda lambda,
             Conversions conversions,
             DiagnosticBag diagnostics,
-            MethodSymbol delegateInvokeMethod,
+            LambdaSymbol lambdaSymbolOpt,
             VariableState initialState,
             ImmutableDictionary<BoundExpression, (NullabilityInfo, TypeSymbol)>.Builder analyzedNullabilityMapOpt,
             ImmutableDictionary<(BoundNode, Symbol), Symbol>.Builder updatedSymbolMapOpt,
+            ReinferredVariableTracker reinferredVariablesOpt,
             SnapshotManager.Builder snapshotBuilderOpt,
             ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> returnTypesOpt)
         {
             Analyze(
                 compilation,
-                lambda.Symbol,
+                lambdaSymbolOpt ?? lambda.Symbol,
```

Comment:
>lambdaSymbolOpt ?? lambda.Symbol [](start = 16, length = 32)

Are there cases when these two a different? If not, consider removing the ```lambdaSymbolOpt``` parameter.

Reply:
Currently, yes, we pass the updated LambdaSymbol through the `VisitConversion` chain.

---
In reply to: [327875276](https://github.com/dotnet/roslyn/pull/38769#discussion_r327875276) [](ancestors = 327875276)